samplePDF/samplePDFFDBase.cpp

```diff
@@ -1414,11 +1358,103 @@ void samplePDFFDBase::addData(TH2D* Data) {
   }
 }
 
+void samplePDFFDBase::SetupNuOscillator() {
+  OscProbCalcerFactory* OscProbCalcFactory = new OscProbCalcerFactory();
+  OscillatorFactory* OscillFactory = new OscillatorFactory();
```

Comment:
OscillFactory is not deleted :(

Reply:
but like...

```c+++
auto OscillFactory = std::make_unique<OscillatorFactory>();
```

is less effort, less characters, more sparkles